solidity/contracts/converter/BancorConverter.sol

```diff
@@ -912,21 +933,27 @@ contract BancorConverter is IBancorConverter, TokenHandler, SmartTokenController
         private
         returns (uint256)
     {
-        uint256[] memory reserveBalances = getBalances(_reserveTokens);
+        syncReserveBalances();
+        reserves[ETH_RESERVE_ADDRESS].balance = reserves[ETH_RESERVE_ADDRESS].balance.sub(msg.value);
+
         IBancorFormula formula = IBancorFormula(addressOf(BANCOR_FORMULA));
-        uint256 supplyAmount = getMinShare(_totalSupply, reserveBalances, _reserveAmounts);
+        uint256 supplyAmount = getMinShare(_totalSupply, _reserveTokens, _reserveAmounts);

```

Comment:
wouldn't it be easier to make sure that _reserveAmounts are already in the correct order by now and use the storage reserveTokens array instead of passing the external one around?

Reply:
@yudilevi : No, it wouldn't. Plus, using the storage array costs more than using a local variable (or in this case, an input argument, which is the same in that sense). We've actually seen this type of optimization at UniSwap and/or Balancer. In this case it is not designated for gas optimization, but reordering the amounts to be in the same order of the reserves? It sounds like `O(N^2)` to me, and even if there's a more efficient way, it's not worth the effort when I already have the **input** reserves in the correct order (because they reflect the user's intention).

Reply:
k